pallets/ddc-verification/src/lib.rs

```diff
@@ -732,11 +755,31 @@ pub mod pallet {
 		///
 		/// Parameters:
 		/// - `leaves`: collection of leaf
-		pub(crate) fn create_merkle_root(leaves: &[ActivityHash]) -> ActivityHash {
-			// todo! add tests
-			let leaves = leaves.iter().map(|l| l.to_vec()).collect::<Vec<Vec<u8>>>();
+		pub(crate) fn create_merkle_root(
+			leaves: &[ActivityHash],
+		) -> Result<ActivityHash, OCWError> {
+			let store = MemStore::default();
+			let mut mmr: MMR<ActivityHash, MergeActivityHash, &MemStore<ActivityHash>> =
+				MemMMR::<_, MergeActivityHash>::new(0, &store);
+			let _ = leaves.iter().map(|a| mmr.push(*a)).collect::<Vec<_>>();
+			let root = mmr.get_root().map_err(|_| OCWError::FailedToCreateMerkleRoot)?;
+
+			Ok(root)
+		}
 
-			T::ActivityHasher::ordered_trie_root(leaves, Default::default()).into()
+		/// Verify whether leaf is part of tree
+		///
+		/// Parameters:
+		/// - `root`: merkle root
+		/// - `leaf`: Leaf of the tree
+		pub(crate) fn proof_merkle_leaf(
+			root: ActivityHash,
+			proof: MerkleProof<ActivityHash, MergeActivityHash>,
+			leaf_with_position: Vec<(u64, ActivityHash)>,
```

Comment:
it hsould be 1 leaf, not vector

if verify wants vector, then just proof.verify(root, vec![leaf_with_position])